File: JavaScript/algos.js
//Palindrome number 
/*
const isPalindrome = (x) => {
    if(x < 0) return false 
    let rev = 0
    let y = x
    while(y>0){
        const lastInt = y % 10
        rev = (rev * 10) + lastInt
        y = (y / 10) | 0
    }
    return x === rev
}
n = 121
// console.log(isPalindrome(n))
*/


// Valid Palindrome (string)
// const isPalindrome = (s) => {
//     return s == s.split('').reverse().join('')
// };
// console.log(isPalindrome("race a car"))
// console.log(isPalindrome("lol"))
// console.log(isPalindrome("amanaplanacanalpanama"))


//FizzBuzz (the most classic lol)
// const fizzBuzz = (n) => {
//     const output = []
//     for(let i = 1; i <= n; i++){
//         if(i % 15 == 0){
//             output.push("FizzBuzz")
//         } else if(i % 3 == 0){
//             output.push("Fizz")
//         } else if(i % 5 == 0){
//             output.push("Buzz") 
//         } else{
//             output.push(i.toString())
//         }
//     }
//     return output
// };
// console.log(fizzBuzz(15))


// Sqrt(x)
// const mySqrt = (x) => {
//     return (Math.floor(Math.sqrt(x)))
// };
// console.log(mySqrt(8))


// Math pow(x,n)
// const myPow = (x, n) => {
//     return x ** n
// };
// console.log(myPow(2,10))



// reverse linked list 
// function reverseList(head){
//     if(head===null) return null 
//     if(head.next===null) return head 
//     let newHead = reverseList(head.next)
//     head.next.next = head 
//     head.next = null 
//     return newHead
// }


//palindrome linked lists 
// function isPalindrome(head){
//     let curr = '';
//     let rev = '';
//     while(head){
//         curr += head.val 
//         rev = head.val + rev
//         head = head.next 
//     }
//     return curr === rev 
// }
// console.log(isPalindrome([1,2,2,1]))

File: Python/algos.py
#FizzBuzz (the most classic lol)
"""
class Solution:
    def fizzBuzz(self, n: int) -> List[str]:
        output = []
        for i in range(1, n + 1):
            if i % 15 == 0:
                output.append("FizzBuzz")
            elif i % 3 == 0:
                output.append("Fizz")
            elif i % 5 == 0:
                output.append("Buzz")
            else:
                output.append(str(i))
        return output
"""
def fizzBuzz(n):
    output = []
    for i in range(1, n + 1):
        if i % 15 == 0:
            output.append("FizzBuzz")
        elif i % 3 == 0:
            output.append("Fizz")
        elif i % 5 == 0:
            output.append("Buzz")
        else:
            output.append(str(i))
    return output
# print(f'FizzBuzz solution for 3 is: {fizzBuzz(3)} ; for 5 is: {fizzBuzz(5)} ; for 15 is: {fizzBuzz(15)}')


#math sqrt(x)
def mySqrt(x):
    return int((x ** 0.5) // 1)
#print(mySqrt(8))


#math pow(x,n)
def myPow(x, n):
    return x ** n
# print(myPow(2,-2))